### PR TITLE
created docstrings for the root colBERT package

### PR DESCRIPTION
**Pull request by epinzur**
these are initial docstrings, created with the help of chatGPT.  We can revise these on another PR.

also made a few parameter name changes, like continuing to rename `passages` to `chunks`

### CODE REVIEW

File: ragstack/colbert/chunk_encoder.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+"""
```
Comment (epinzur) [inline ragstack/colbert/chunk_encoder.py:1]: new version of `passage_encoder.py`

File: pyproject.toml [hunk 2]
```diff
@@ -27,6 +27,8 @@ langchain-nvidia-ai-endpoints = { version = "0.0.3", optional = true }
 colbert-ai = { version = "0.2.19", optional = true }
 pyarrow = { version = "14.0.1", optional = true }
 torch = { version = "2.2.1", optional = true }
+faiss-gpu = { version = ">=1.7.0", optional = true }
```
Comment (zzzming) [inline pyproject.toml:30]: Can we just lock on a version? Faiss can have compatibility issues if it's not tested.

**Review by zzzming — APPROVED** [review]
I don't think we need faiss as we discussed

File: pyproject.toml (hunk 2 above)
Comment (epinzur) [reply pyproject.toml]: i just removed it from the dependency list

**Pull request merged**